app/src/main/java/com/example/composetemplate/repositories/AuthDataSource.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.composetemplate.repositories
+
+import com.example.composetemplate.data.models.local_models.User
+import com.example.composetemplate.data.remote.base.BaseRequest
+import com.example.composetemplate.utils.LoginCallback
+import com.example.composetemplate.utils.SuccessCallback
+
+/** In this interface we set the all actions of auth and then we implement them in each service we want to use.
+ * that Way we make our code more flexible and changeable and easily can change the all service we use
+ * using the DI
+ *
+ * */
+
+sealed interface AuthDataSource {
```

Comment:
Please rename this class. An interface shouldn't be called DataSource. Consider renaming it to something like Authable, Authenticatable, AuthActionable, or similar.